testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/chat/ContactSelectionWindow.kt

```diff
@@ -0,0 +1,102 @@
+
+    private lateinit var contactSelectionAdapter: ContactSelectionAdapter
+    private lateinit var contactSelectionLayout: View
+
+    private val hasContact =
+        AppSettings.instance.isContactsSaved && !AppSettings.instance.contacts.isNullOrEmpty()
+
+    private lateinit var message: MessageToContact
+    private lateinit var onSuccessSingleContact: (contactId: String?) -> Unit
+    private lateinit var onErrorSingleContact: (message: String) -> Unit
+
+    fun openSingleContactSelection(
+        message: MessageToContact,
+        onSuccess: (contactId: String?) -> Unit,
+        onError: (message: String) -> Unit
+    ) {
+        if (!hasContact) {
+            showAlertDialog(
+                activity,
+                "Warning",
+                "There is no contact found saved in HostApp."
+            )
+            return
+        }
+
+        this.message = message
+        this.onSuccessSingleContact = onSuccess
+        this.onErrorSingleContact = onError
+
+        // initialize contact selection view
+        initDefaultWindow()
+        prepareDataForAdapter()
+
+        // preview dialog
+        contactSelectionAlertDialog.show()
+    }
+
+    private fun initDefaultWindow() {
+        val layoutInflater = LayoutInflater.from(activity)
+        contactSelectionLayout = layoutInflater.inflate(R.layout.window_contact_selection, null)
+        val recyclerView =
+            contactSelectionLayout.findViewById<RecyclerView>(R.id.listContactSelection)
+        recyclerView.layoutManager = LinearLayoutManager(activity)
+        recyclerView.addItemDecoration(
+            DividerItemDecoration(activity, DividerItemDecoration.VERTICAL)
+        )
+
+        contactSelectionAdapter = ContactSelectionAdapter(this)
+        recyclerView.adapter = contactSelectionAdapter
+
+        contactSelectionAlertDialog =
+            AlertDialog.Builder(activity, R.style.AppTheme_DefaultWindow).create()
+        contactSelectionAlertDialog.setView(contactSelectionLayout)
+        contactSelectionLayout.findViewById<ImageView>(R.id.contactCloseWindow).setOnClickListener {
+            contactSelectionAlertDialog.dismiss()
+        }
+    }
+
+    private fun prepareDataForAdapter() {
+        if (hasContact) contactSelectionAdapter.addContactList(AppSettings.instance.contacts)
+    }
+
+    override fun onContactSelect(contact: Contact) {
+        when {
+            message.isEmpty -> onErrorSingleContact("The message sent was empty.")
+            contact.id.isEmpty() -> {
+                onErrorSingleContact("There is no contact found in HostApp.")
+            }
+            else -> {
+                onSuccessSingleContact(contact.id)
+                // Note: Doesn't need to actually send a message because we don't have an interface for this in the demo app.
+                showAlertDialog(
+                    activity,
+                    "Success!",
+                    "The message has been sent to contact id: ${contact.id}"
```

Comment:
Could we also display the message data in this alert? i.e. the text, image, etc.

Reply:
Yeah, I have updated this part.